backend/spherre/app/service/transaction.py

```
@@ -0,0 +1,297 @@
+        ).one_or_none()
+        
+        if not transaction:
+            raise ValueError(f"Transaction with ID {transaction_id} not found for this account")
+            
+        # Verify member is part of the account
+        if member not in account.members:
+            raise ValueError("Member must belong to the account")
+            
+        # Check if transaction is in an approvable state
+        if transaction.status != TransactionStatus.INITIATED:
+            raise ValueError(f"Transaction is in {transaction.status} state and cannot be approved")
```

Ruff (0.11.9)

Line too long (100 > 88)

(E501)